### PR TITLE
DOC: Add plot to circmean docstring

### PR DESCRIPTION
**Pull request by dschmitz89**
#### What does this implement/fix?
Include a plot of circular data and a comparison of circular and regular mean in the docstring of `stats.circmean`

#### Additional information
I think for many users the functionality under circular/directional statistics is not immediately clear. In case of `circmean`, the usefulness can very easily be demonstrated by a plot of circular data and their circular/regular mean values.

### CODE REVIEW

File: scipy/stats/_morestats.py [hunk 1]
```diff
@@ -3741,14 +3741,31 @@ def circmean(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
...
+
+    >>> plt.plot(np.cos(np.linspace(0, 2*np.pi, 500),
+    ...          np.sin(np.linspace(0, 2*np.pi, 500),
+    ...          c='k')
+    >>> plt.scatter(np.cos(angles) , np.sin(angles), c='k')
+    >>> plt.scatter(np.cos(circmean), np.sin(circmean), c='b',
+    ...             label='circmean')
+    >>> plt.scatter(np.cos(mean), np.sin(mean), c='r', label='mean')
+    >>> plt.legend()
+    >>> plt.show()
```
Comment (mdhaber) [inline scipy/stats/_morestats.py:3768]: The refguide failure looks real:
```
scipy.stats.circmean
--------------------

File "build/testenv/lib/python3.9/site-packages/scipy/stats/_morestats.py", line 3760, in circmean
Failed example:
    plt.plot(np.cos(np.linspace(0, 2*np.pi, 500),
             np.sin(np.linspace(0, 2*np.pi, 500),
             c='k')
Exception raised:
```

File: scipy/stats/_morestats.py [hunk 2]
```diff
@@ -3741,14 +3741,31 @@ def circmean(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
...
+    7.294976657784009
+
+    >>> mean=angles.mean()
+    >>> np.rad2deg(mean)
+    126.66666666666666
+
+    Plot and compare the results of *circmean* and the regular *mean*.
+
+    >>> plt.plot(np.cos(np.linspace(0, 2*np.pi, 500),
+    ...          np.sin(np.linspace(0, 2*np.pi, 500),
```
Comment (mdhaber) [inline scipy/stats/_morestats.py:3761]:
```suggestion
    >>> plt.plot(np.cos(np.linspace(0, 2*np.pi, 500)),
    ...          np.sin(np.linspace(0, 2*np.pi, 500)),
```

File: scipy/stats/_morestats.py [hunk 3]
```diff
@@ -3741,14 +3741,31 @@ def circmean(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
...
     >>> from scipy.stats import circmean
-    >>> circmean([0.2, 1.4, 2.6], high = 1, low = 0)
-    0.4
+    >>> import matplotlib.pyplot as plt
+    >>> angles=np.deg2rad(np.array([20, 30, 330]))
+    >>> circmean=circmean(angles)
+    >>> np.rad2deg(circmean)
+    7.294976657784009
+
+    >>> mean=angles.mean()
```
Comment (mdhaber) [inline scipy/stats/_morestats.py:3754]: We should follow PEP8 standards in docstring examples, too.
```suggestion
    >>> angles = np.deg2rad(np.array([20, 30, 330]))
    >>> circmean = circmean(angles)
    >>> np.rad2deg(circmean)
    7.294976657784009

    >>> mean = angles.mean()
```

File: scipy/stats/_morestats.py [hunk 4]
```diff
@@ -3741,14 +3741,31 @@ def circmean(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
...
+    >>> np.rad2deg(mean)
+    126.66666666666666
+
+    Plot and compare the results of *circmean* and the regular *mean*.
+
+    >>> plt.plot(np.cos(np.linspace(0, 2*np.pi, 500)),
+    ...          np.sin(np.linspace(0, 2*np.pi, 500)),
+    ...          c='k')
+    >>> plt.scatter(np.cos(angles) , np.sin(angles), c='k')
```
Comment (mdhaber) [inline scipy/stats/_morestats.py:3763]:
```suggestion
    >>> plt.scatter(np.cos(angles), np.sin(angles), c='k')
```

File: scipy/stats/_morestats.py [hunk 5]
```diff
@@ -3741,14 +3741,31 @@ def circmean(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
...
+    >>> angles=np.deg2rad(np.array([20, 30, 330]))
+    >>> circmean=circmean(angles)
+    >>> np.rad2deg(circmean)
+    7.294976657784009
+
+    >>> mean=angles.mean()
+    >>> np.rad2deg(mean)
+    126.66666666666666
+
+    Plot and compare the results of *circmean* and the regular *mean*.
```
Comment (mdhaber) [inline scipy/stats/_morestats.py:3758]: Asterisks are emphasis. How about:
```suggestion
    Plot and compare the circular mean against the arithmetic mean.
```

File: scipy/stats/_morestats.py [hunk 6]
```diff
@@ -3741,14 +3741,31 @@ def circmean(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
...
+
+    >>> plt.plot(np.cos(np.linspace(0, 2*np.pi, 500)),
+    ...          np.sin(np.linspace(0, 2*np.pi, 500)),
+    ...          c='k')
+    >>> plt.scatter(np.cos(angles) , np.sin(angles), c='k')
+    >>> plt.scatter(np.cos(circmean), np.sin(circmean), c='b',
+    ...             label='circmean')
+    >>> plt.scatter(np.cos(mean), np.sin(mean), c='r', label='mean')
+    >>> plt.legend()
+    >>> plt.show()
```
Comment (mdhaber) [inline scipy/stats/_morestats.py:3768]:
```suggestion
    >>> plt.legend()
    >>> plt.axis('equal')
    >>> plt.show()
```

**Comment by mdhaber** [comment]
OK @dschmitz89. I corrected the docstring example by adding the required closing parentheses. Please test examples locally.

If the other suggestions look good to you, please commit and I'll merge when CI finishes.

**Comment by dschmitz89** [comment]
> OK @dschmitz89. I corrected the docstring example by adding the required closing parentheses. Please test examples locally.
> 
> If the other suggestions look good to you, please "Add suggestions to batch" them commit them all together. I'll merge when CI finishes.

Thanks for correcting, just applid your suggestions. I would have loved to test locally but building docs does not work on my machine somehow.

**Comment by mdhaber** [comment]
> building docs does not work on my machine somehow.

Yeah, me neither. I just mean to copy-paste them into a prompt.

Thanks @dschmitz89!

**Pull request merged**